internal/cluster/cluster.go

```diff
@@ -522,32 +524,52 @@ func (m *Manager) ClusterMonitoring() {
 		if len(clusters) == 0 {
 			break
 		}
-		for _, cluster := range clusters {
-			if !m.leaderElector.IsLeader() {
-				m.log.Debugf("Not a leader, exiting ClusterMonitoring")
-				return
-			}
-			if !m.SkipMonitoring(cluster) {
 
-				if err = m.setConnectivityMajorityGroupsForClusterInternal(cluster, m.db); err != nil {
-					log.WithError(err).Error("failed to set majority group for clusters")
-				}
-				clusterAfterRefresh, err = m.refreshStatusInternal(ctx, cluster, m.db)
-				if err != nil {
-					log.WithError(err).Errorf("failed to refresh cluster %s state", cluster.ID)
-					continue
-				}
-
-				if swag.StringValue(clusterAfterRefresh.Status) != swag.StringValue(cluster.Status) {
-					log.Infof("cluster %s updated status from %s to %s via monitor", cluster.ID,
-						swag.StringValue(cluster.Status), swag.StringValue(clusterAfterRefresh.Status))
-				}
+		jobs := make(chan *common.Cluster, limit)
+		for i := range clusters {
+			jobs <- clusters[i]
+		}
 
-				if m.shouldTriggerLeaseTimeoutEvent(cluster, curMonitorInvokedAt) {
-					m.triggerLeaseTimeoutEvent(ctx, cluster)
+		var wg sync.WaitGroup
+		for i := 0; i < m.MonitorWorkersNum; i++ {
+			wg.Add(1)
```

Comment:
Why not have the workers be alive during the whole monitoring cycle?

Reply:
The monitoring function is just a function after all. it is being called every x seconds and it's not a long-living entity. 
I can probably change the design, it's a bigger change but not impossible of course.